code/app/src/main/java/com/example/bread/fragment/HistoryFragment.java

```diff
@@ -133,4 +146,99 @@ public void onSuccess(Void aVoid) {
         }
         selectedEvents.clear();  // Clear the selection after deletion
     }
+
+    //https://stackoverflow.com/questions/17210839/get-last-week-date-range-for-a-date-in-java
+
+    /**
+     * Retrieves a date range for the most recent week (7 days).
+     * The range starts from 7 days ago and ends at the current date.
+     *
+     * @return An ArrayList containing two Date objects:
+     *         the start date (7 days ago) and the end date (today).
+     */
+    public ArrayList<Date> getMostRecentWeek(){
+        ArrayList<Date> weekRange = new ArrayList<Date>();
+        Date date = new Date(); //gets current date
+        Calendar c = Calendar.getInstance(); //sets calendar to today
+        c.setTime(date);
+        Date end = c.getTime();
+        c.add(Calendar.DATE, -7);
+        Date start = c.getTime();
+
+        weekRange.add(start);
+        weekRange.add(end);
+
+        return weekRange;
+    }
+
+    /**
+     * Filters the displayed mood events to show only those from the most recent week,
+     * if the filter switch is enabled.
+     * If the switch is off, all mood events are reloaded.
+     *
+     * @param isChecked True if the "Most Recent Week" filter is enabled, false otherwise.
+     */
+    @Override
+    public void mostRecentWeek(boolean isChecked) {
+        if (isChecked){
+            moodsRepo.listenForEventsWithParticipantRef(participantRef, moodEvents -> {
+                        if (moodEvents != null) {
+                            //create functionality so it filters for moods in last 7 days
+                            moodEventArrayList.clear();
+                            ArrayList<Date> weekRange = getMostRecentWeek();
+                            for (int i = 0; i < moodEvents.size(); i++){
+                                if ((moodEvents.get(i).getTimestamp().before(weekRange.get(0))) //if mood is in last week
+                                    &&
+                                    (moodEvents.get(i).getTimestamp().after(weekRange.get(1)))){
+                                    moodEventArrayList.add(moodEvents.get(i));
+                                }
+                            }
+                            moodEventArrayList.sort((e1, e2) -> e2.compareTo(e1));
+                        }
+                        moodArrayAdapter.notifyDataSetChanged();
+                    },
+                    error -> {
+                        Log.e("History Fragment", "Failed to listen for mood events", error);
+                    });
+        }
+        else{ //!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!??????? might mess with other filters? call it first?
+            //call loadMoodEvents() to keep things the same
+            loadMoodEvents();
```

Comment:
A user could be allowed to apply all the filters so it would make more sense to way that allows users to select multiple options and apply those filters. 

For now, it's okay but we could add a List<Interface> that could contain the selected filters and call those filters to be applied on the fetched data when the user selects filter button. Just a thought but could do in a different way as well.